### PR TITLE
allow packaging >=22.0,<23.0

### PR DESCRIPTION
**Pull request by ellisvalentiner**
## Description

Loosen dependency on packaging from 22.0 to <23.0

## Motivation and Context

This change will allow prospector to be installed alongside other common tools (e.g. black) use packaging >22.0.

## How Has This Been Tested?

This change was tested by running the pre-commit hooks and CI workflow.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] My change requires a change to the dependencies
- [x] I have updated the dependencies accordingly
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by Pierre-Sassoulas — COMMENTED** [review]
Imo the dependency should be >= something to be compatible with the max number of over repositories. We seems to support everything from 0 to 22.x...

**Comment by ellisvalentiner** [comment]
Because of the dependency on `requirements-detector`, which requires `packaging` >=21.3, I set the lower bound to the same.

**Comment by Pierre-Sassoulas** [comment]
Shouldn't we let ``requirements-detector`` set it's own constraint and keep our constraint large or apply constraint on ``requirements-detector`` ? Do prospector actually needs "packaging" or is this just a transitive dependency brought by  ``requirements-detector`` ?

**Comment by ellisvalentiner** [comment]
Per [this comment](https://github.com/PyCQA/prospector/pull/576#pullrequestreview-1241147422) on a previous PR, it seems that the version for `requirements-detector` should be set explicitly.

**Comment by Pierre-Sassoulas** [comment]
Huhuhu, right, let's listen to this guy he seems to knows what he's talking about.

**Review by Pierre-Sassoulas — APPROVED** [review]

**Review by Pierre-Sassoulas — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -60,7 +60,7 @@ mypy = {version = ">=0.600", optional = true}
 pyroma = {version = ">=2.4", optional = true}
 setoptconf-tmp = "^0.3.1"
 GitPython = "^3.1.27"
-packaging = "<22.0"
+packaging = ">=21.0,<23.0"
```
[the remark this answers is not in the document]
Comment (carlio) [reply pyproject.toml]: I'm looking into fixing the requirements-detector package from which this transitive dependency comes. However I have run into [downstream issues](https://github.com/landscapeio/requirements-detector/issues/40) with poetry-semver and python-semver so it'll take me a little while to get a decent release of requirements-detector out.

**Comment by carlio** [comment]
I have just released requirements-detector version 1.1.0 which has a much more lax constraint. I'm not sure why it was set to the range it was, I forget exactly. Anyway, now it won't be an issue for prospector.

**Pull request merged**